pandas/core/arrays/base.py

```diff
@@ -354,7 +355,7 @@ def ndim(self) -> int:
         """
         Extension Arrays are only allowed to be 1-dimensional.
         """
-        return 1
+        return len(self.shape)
```

Comment:
I am not sure this change will help. There can be EAs out there that define this themselves and override this with a hardcoded 1, so we will still need to define a wrapper I think? 
(so therefore I would maybe rather leave it as is, to ensure we cover that use case)

Reply:
im not sure I understand the problem here.  is there a case in which this wont be correct?

Reply:
I supposed you made this change because for the other PR you are patching `self.shape` to return (N, 1) or (1, N), and so with this change, ndim automatically follows that. 
But you can't rely on the fact that self.ndim already is correctly following self.shape, so you will always have to patch ndim as well. 

(exact terminology of "patching" might not be fully reflect to other PR on 2D EAs, need to update myself on that, but hope to give the idea)

Reply:
will revert so we'll handle it in the next pass